src/chainstate/burn/db/burndb.rs

```diff
@@ -913,6 +915,24 @@ impl BurnDB {
         }
     }
     
+    /// Get a snapshot for an existing burn chain block given its consensus hash.
+    pub fn get_block_snapshot_consensus(conn: &Connection, consensus_hash: &ConsensusHash) -> Result<Option<BlockSnapshot>, db_error> {
+        let qry = "SELECT * FROM snapshots WHERE consensus_hash = ?1".to_string();
+        let args = [&consensus_hash];
+        let rows = query_rows::<BlockSnapshot, _>(conn, &qry.to_string(), &args)?;
+        match rows.len() {
+            0 => {
+                test_debug!("No snapshot with consensus hash {}", consensus_hash);
+                Ok(None)
+            },
+            1 => Ok(Some(rows[0].clone())),
```

Comment:
This clone should be avoided with a `pop()` or similar.

Reply:
Fixed in 48a05b2c0